src/probability/integration.lean

```diff
@@ -118,4 +117,127 @@ lintegral_mul_eq_lintegral_mul_lintegral_of_independent_measurable_space
   (measurable_iff_comap_le.1 h_meas_f) (measurable_iff_comap_le.1 h_meas_g) h_indep_fun
   (measurable.of_comap_le le_rfl) (measurable.of_comap_le le_rfl)
 
+/-- If `f` and `g` with values in `ℝ≥0∞` are independent and almost everywhere measurable,
+   then `E[f * g] = E[f] * E[g]` (slightly generalizing
+   `lintegral_mul_eq_lintegral_mul_lintegral_of_indep_fun`). -/
+lemma lintegral_mul_eq_lintegral_mul_lintegral_of_indep_fun' [measurable_space α] {μ : measure α}
+  {f g : α → ℝ≥0∞} (h_meas_f : ae_measurable f μ) (h_meas_g : ae_measurable g μ)
+  (h_indep_fun : indep_fun f g μ) :
+  ∫⁻ a, (f * g) a ∂μ = ∫⁻ a, f a ∂μ * ∫⁻ a, g a ∂μ :=
+begin
+  have fg_ae : f * g =ᵐ[μ] (h_meas_f.mk _) * (h_meas_g.mk _),
+    from h_meas_f.ae_eq_mk.mul h_meas_g.ae_eq_mk,
+  rw [lintegral_congr_ae h_meas_f.ae_eq_mk, lintegral_congr_ae h_meas_g.ae_eq_mk,
+    lintegral_congr_ae fg_ae],
+  apply lintegral_mul_eq_lintegral_mul_lintegral_of_indep_fun
+    h_meas_f.measurable_mk h_meas_g.measurable_mk,
+  exact h_indep_fun.ae_eq h_meas_f.ae_eq_mk h_meas_g.ae_eq_mk
+end
+
+/-- The product of two independent, integrable, real_valued random variables is integrable. -/
+lemma indep_fun.integrable_mul {mα : measurable_space α} {μ : measure α}
+  {β : Type*} {mβ : measurable_space β} [normed_division_ring β] [borel_space β]
+  {X Y : α → β} (hXY : indep_fun X Y μ) (hX : integrable X μ) (hY : integrable Y μ) :
+  integrable (X * Y) μ :=
+begin
+  let nX : α → ennreal := λ a, ∥X a∥₊,
+  let nY : α → ennreal := λ a, ∥Y a∥₊,
+
+  have hXY' : indep_fun (λ a, ∥X a∥₊) (λ a, ∥Y a∥₊) μ :=
+    hXY.comp measurable_nnnorm measurable_nnnorm,
+  have hXY'' : indep_fun nX nY μ :=
+    hXY'.comp measurable_coe_nnreal_ennreal measurable_coe_nnreal_ennreal,
+
+  have hnX : ae_measurable nX μ := hX.1.ae_measurable.nnnorm.coe_nnreal_ennreal,
+  have hnY : ae_measurable nY μ := hY.1.ae_measurable.nnnorm.coe_nnreal_ennreal,
+
+  have hmul : ∫⁻ a, nX a * nY a ∂μ = ∫⁻ a, nX a ∂μ * ∫⁻ a, nY a ∂μ :=
+    by convert lintegral_mul_eq_lintegral_mul_lintegral_of_indep_fun' hnX hnY hXY'',
+
+  refine ⟨hX.1.mul hY.1, _⟩,
+  simp_rw [has_finite_integral, pi.mul_apply, nnnorm_mul, ennreal.coe_mul, hmul],
+  exact ennreal.mul_lt_top_iff.mpr (or.inl ⟨hX.2, hY.2⟩)
+end
+
+/-- The (Bochner) integral of the product of two independent, nonnegative random
+  variables is the product of their integrals. The proof is just plumbing around
+  `lintegral_mul_eq_lintegral_mul_lintegral_of_indep_fun'`. -/
+lemma indep_fun.integral_mul_of_nonneg [measurable_space α] {μ : measure α} {X Y : α → ℝ}
```

Comment:
could you make the measurable_space argument implicit here and in the next one as well? Actually since you use the `measurable_space`, `measure` and the two functions `X` and `Y` in all three lemmas, it would be better to define them only once with `variables`

Reply:
OK I missed those. I took the liberty of doing the same thing with the previous lemmas (which I did not write) and factoring all the variables to the beginning of the file.